router/core/router.go

```diff
@@ -588,9 +619,17 @@ func (r *Router) newServer(ctx context.Context, routerConfig *nodev1.RouterConfi
 
 		subgraphs := make([]Subgraph, len(routerConfig.Subgraphs))
 		for _, s := range routerConfig.Subgraphs {
-			parsedURL, err := url.Parse(s.RoutingUrl)
+			rawURL := s.RoutingUrl
```

Comment:
Move the validation logic into the proposed `configureSubgraphOverwrites` and return the error when a subgraph has an invalid url. Although the URLs are already validated through the validate config package. I'm also fine with both.